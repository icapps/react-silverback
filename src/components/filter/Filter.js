import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { strings } from '../../utils';
import './filter.scss';

const search = require('../../assets/images/search.svg');

const Filter = ({ handleFilter }) => {
  const [filterValue, setFilterValue] = useState('');
  const handleChange = ({ target: { value } }) => {
    setFilterValue(value);
    handleFilter(value);
  };
  return (
    <div className="input-group filter">
      <div className="input-group-prepend">
        <span className="input-group-text">
          <img src={search} alt={strings.SEARCH} />
        </span>
      </div>
      <input
        className="form-control"
        type="text"
        value={filterValue}
        onChange={handleChange}
        placeholder={strings.SEARCH}
      />
    </div>
  );
};

Filter.propTypes = {
  handleFilter: PropTypes.func.isRequired,
};

export default Filter;
