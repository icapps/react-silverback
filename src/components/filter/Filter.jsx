import React from 'react';
import PropTypes from 'prop-types';
import { strings } from '../../utils';
import './filter.css';

const search = require('../../assets/images/search.svg');

class Filter extends React.Component {
  constructor(props) {
    super(props);
    this.state = { filterValue: '' };
  }

  handleChange = event => {
    this.setState({ filterValue: event.target.value }, this.props.handleFilter(event.target.value));
  }

  render() {
    return (
      <div className="input-group">
        <div className="input-group-prepend">
          <span className="input-group-text"><img src={search} alt={strings.SEARCH} /></span>
        </div>
        <input className="form-control filter" type="text" value={this.state.filterValue} onChange={this.handleChange} placeholder={strings.SEARCH} />
      </div>
    );
  }
}

Filter.propTypes = {
  handleFilter: PropTypes.func.isRequired,
};

export default Filter;
