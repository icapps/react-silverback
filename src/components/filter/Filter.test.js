import React from 'react';
import { shallow, configure } from 'enzyme';
import Filter from './Filter';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Filter Component', () => {
  it('should render a Filter component', () => {
    const wrapper = shallow(<Filter handleFilter={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a Filter component that can handle change', () => {
    const wrapper = shallow(<Filter handleFilter={() => { }} />);
    wrapper.find('input').simulate('change', { target: { value: 'test' } });
    expect(wrapper).toMatchSnapshot();
  });
});
