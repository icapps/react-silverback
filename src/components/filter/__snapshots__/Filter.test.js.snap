// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component should render a Filter component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filter
    handleFilter={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="input-group-prepend"
>
          <span
                    className="input-group-text"
          >
                    <img
                              alt="Search"
                              src="search.svg"
                    />
          </span>
</div>,
        <input
          className="form-control filter"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
/>,
      ],
      "className": "input-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="input-group-text"
>
            <img
                        alt="Search"
                        src="search.svg"
            />
</span>,
          "className": "input-group-prepend",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Search"
              src="search.svg"
/>,
            "className": "input-group-text",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Search",
              "src": "search.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control filter",
          "onChange": [Function],
          "placeholder": "Search",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="input-group-prepend"
>
            <span
                        className="input-group-text"
            >
                        <img
                                    alt="Search"
                                    src="search.svg"
                        />
            </span>
</div>,
          <input
            className="form-control filter"
            onChange={[Function]}
            placeholder="Search"
            type="text"
            value=""
/>,
        ],
        "className": "input-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="input-group-text"
>
              <img
                            alt="Search"
                            src="search.svg"
              />
</span>,
            "className": "input-group-prepend",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Search"
                src="search.svg"
/>,
              "className": "input-group-text",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Search",
                "src": "search.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control filter",
            "onChange": [Function],
            "placeholder": "Search",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Filter Component should render a Filter component that can handle change 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filter
    handleFilter={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="input-group-prepend"
>
          <span
                    className="input-group-text"
          >
                    <img
                              alt="Search"
                              src="search.svg"
                    />
          </span>
</div>,
        <input
          className="form-control filter"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value="test"
/>,
      ],
      "className": "input-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="input-group-text"
>
            <img
                        alt="Search"
                        src="search.svg"
            />
</span>,
          "className": "input-group-prepend",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Search"
              src="search.svg"
/>,
            "className": "input-group-text",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Search",
              "src": "search.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control filter",
          "onChange": [Function],
          "placeholder": "Search",
          "type": "text",
          "value": "test",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="input-group-prepend"
>
            <span
                        className="input-group-text"
            >
                        <img
                                    alt="Search"
                                    src="search.svg"
                        />
            </span>
</div>,
          <input
            className="form-control filter"
            onChange={[Function]}
            placeholder="Search"
            type="text"
            value="test"
/>,
        ],
        "className": "input-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="input-group-text"
>
              <img
                            alt="Search"
                            src="search.svg"
              />
</span>,
            "className": "input-group-prepend",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Search"
                src="search.svg"
/>,
              "className": "input-group-text",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Search",
                "src": "search.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control filter",
            "onChange": [Function],
            "placeholder": "Search",
            "type": "text",
            "value": "test",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
